build/seqan-config.cmake

```diff
@@ -0,0 +1,364 @@
+    return ()
+endif ()
+
+# ----------------------------------------------------------------------------
+# System dependencies
+# ----------------------------------------------------------------------------
+
+# librt
+if ((${CMAKE_SYSTEM_NAME} STREQUAL "Linux") OR
+    (${CMAKE_SYSTEM_NAME} STREQUAL "kFreeBSD") OR
+    (${CMAKE_SYSTEM_NAME} STREQUAL "GNU"))
+    set (SEQAN_LIBRARIES ${SEQAN_LIBRARIES} rt)
+endif ()
+
+# libpthread
+if (UNIX AND NOT APPLE)
+    # pthread flag (without -l) also defines macros on some platforms
+    set (SEQAN_CXX_FLAGS "${SEQAN_CXX_FLAGS} -pthread")
+    # need to explicitly link lpthread with whole-archive because Debian strips bits from the library
+    # that are required for C++14 threading, resulting in broken static builds
+    set (SEQAN_LIBRARIES ${SEQAN_LIBRARIES} "-Wl,--whole-archive -lpthread -Wl,--no-whole-archive")
+endif ()
+
+# libexecinfo -- implicit
+check_include_file_cxx(execinfo.h _SEQAN_HAVE_EXECINFO)
+mark_as_advanced(_SEQAN_HAVE_EXECINFO)
+if (_SEQAN_HAVE_EXECINFO)
+    if ((${CMAKE_SYSTEM_NAME} STREQUAL "FreeBSD") OR (${CMAKE_SYSTEM_NAME} STREQUAL "OpenBSD"))
+        set (SEQAN_LIBRARIES ${SEQAN_LIBRARIES} execinfo elf)
+    endif ()
+endif (_SEQAN_HAVE_EXECINFO)
+
+# ----------------------------------------------------------------------------
+# ZLIB dependency
+# ----------------------------------------------------------------------------
+
+if (NOT SEQAN3_NO_ZLIB)
+    find_package (ZLIB ${SeqAn_FIND_QUIETLY})
+endif ()
+
+if (ZLIB_FOUND)
+    set (SEQAN_LIBRARIES         ${SEQAN_LIBRARIES}         ${ZLIB_LIBRARIES})
+    set (SEQAN_INCLUDE_DIRS      ${SEQAN_INCLUDE_DIRS}      ${ZLIB_INCLUDE_DIRS})
+    set (SEQAN_DEFINITIONS       ${SEQAN_DEFINITIONS}       "-DSEQAN3_HAS_ZLIB=1")
+endif ()
+
+# ----------------------------------------------------------------------------
+# BZip2 dependency
+# ----------------------------------------------------------------------------
+
+if (NOT SEQAN3_NO_BZIP2)
+    find_package (BZip2 ${SeqAn_FIND_QUIETLY})
+endif ()
+
+if (NOT ZLIB_FOUND AND BZIP2_FOUND)
+    # NOTE(marehr): iostream_bzip2 uses the type `uInt`, which is defined by
+    # `zlib`. Therefore, `bzip2` will cause a ton of errors without `zlib`.
+    message (AUTHOR_WARNING "Disabling BZip2 [which was successfully found], "
+                            "because ZLIB was not found. BZip2 depends on ZLIB.")
+    unset (BZIP2_FOUND)
+endif ()
+
+if (BZIP2_FOUND)
+    set (SEQAN_LIBRARIES         ${SEQAN_LIBRARIES}         ${BZIP2_LIBRARIES})
+    set (SEQAN_INCLUDE_DIRS      ${SEQAN_INCLUDE_DIRS}      ${BZIP2_INCLUDE_DIRS})
+    set (SEQAN_DEFINITIONS       ${SEQAN_DEFINITIONS}       "-DSEQAN3_HAS_BZIP2=1")
+endif ()
+
+# ----------------------------------------------------------------------------
+# Final check of compilability of seqan3/core/platform.hpp
+# ----------------------------------------------------------------------------
+
+#TODO
```

Comment:
this will compile an example against the platform.hpp which contains more checks.